test/e2e/crud.go

```diff
@@ -92,68 +91,76 @@ var _ = Describe("Federated types", func() {
 
 			// TODO(marun) Is there a better way to create crd's from code?
 
-			crdKind := "FedTestCrd"
+			targetCrdKind := "FedTestCrd"
```

Comment:
There are only 2 nested describes in this file, so I'm not sure what you mean.  How would you propose the tests be structured?

Reply:
It's more of a combination of all the describe containers, it specs, and anonymous functions that seems to be hurting readability at first glance. Once you understand how it's broken up it makes sense. I haven't thought much about it and don't know if there is a Ginkgo best practice for this. Some thoughts that come to mind:

- It may be possible to avoid too many nested anonymous functions and replace the anonymous functions with identifiers for those functions defined elsewhere. However, this could be problematic with the use of function closures.
- The 2 nested describes appear to be sharing the federation framework, e2e logger, and namespaceTypeConfig. Perhaps there's a better way to share this context amongst the tests.
- I think the CRD Describe container is probably the one that is getting a bit long. I think focusing on this would probably help the most such as refactoring it into separate functions that can possibly be re-used. Whenever we start generating the CRDs, it can probably be refactored too.

Anyway, not sure how valid any of those ideas are without investigating further. For now we have higher priority stuff.